src/graphql-types/ReasonApolloMutation.re

```diff
@@ -55,16 +53,14 @@ module MutationFactory = (Config:Config) => {
     let apolloDataToReason: renderPropObjJS => response = 
       apolloData =>
         switch (
-          apolloData##called |> Js.to_bool,
           apolloData##loading |> Js.to_bool,
           apolloData##data |> ReasonApolloUtils.getNonEmptyObj,
           apolloData##error |> Js.Nullable.to_opt
         ) {
-        | (true, false, _, _) => Called
-        | (_, true, _, _) => Loading
-        | (false, false, Some(data), None) => Data(Config.parse(data))
-        | (false, false, _, Some(error)) => Error(error)
-        | (false, false, None, None) => NoData
+        | (true, _, _) => Loading
+        | (false, Some(data), _) => Data(Config.parse(data))
+        | (false, _, Some(error)) => Error(error)
+        | (false, None, None) => Error({"message": "No data", "graphQLErrors": Js.Nullable.null, "networkError": Js.Nullable.null})
```

Comment:
Hmm, arent’t those pattern matching still wrong? `called` is `false` only initially, then no matter what the result is it will always be `true`. So none of those will ever match.

That was what my PR was aiming to fix (besides the additional NotCalled variant). #82

Reply:
It's working great for me now as it's not matching on `apolloData##called` anymore... it's only matching on `loading`, `data`, and `error`.

We are in an "Error" state before calling the mutation so it does require a bit more work when rendering server errors.  Just need to avoid rendering an error when `graphQLErrors` and `networkError` are both null since the user hasn't submitted yet.  Would be easier matching on the NoData variant but understand the desire to keep the variant small.

Reply:
@emmenko  as @tmepple mentionned, it's not pattern matching on `called` anymore.
I added a fix https://github.com/apollographql/reason-apollo/commit/c2c45e69f642a22a52ef4c16bdc9ae1c6d872c4c that added an initial state `NotCalled`. Called means it is Error or Data

Reply:
Just published under `reason-apollo@beta` (0.9.0)

Reply:
Oh, my bad sorry! I somehow missed that the first argument `called` was removed

<img width="387" alt="image" src="https://user-images.githubusercontent.com/1110551/40332375-33943dbe-5d54-11e8-9e57-28e00f56911b.png">

I should watch more carefully next time 🙃

Alright, all good then, thanks again for making this simpler!

Reply:
Working well for me now, great job thanks!